### PR TITLE
feat: Implement translate article - EXO-72085 - Meeds-io/MIPs#128

### PR DESCRIPTION
**Pull request by sofyenne**

### CODE REVIEW

File: content-service/src/main/java/io/meeds/news/rest/NewsRest.java [hunk 1]
```diff
@@ -784,4 +785,40 @@ private NewsFilter buildFilter(List<String> spaces, String filter, String text,
...
+      if (StringUtils.isBlank(id)) {
+        return Response.status(Response.Status.BAD_REQUEST).build();
+      }
+      // fetch always the original news
+      News news = newsService.getNewsById(id, currentIdentity, false, ARTICLE.name());
+      if (news == null) {
+        return Response.status(Response.Status.NOT_FOUND).build();
+      }
+      if (!news.isCanDelete()) {
+        throw new IllegalAccessException("User " + currentIdentity.getUserId() + " is not authorized to delete article translation");
```
Comment (ahamdi) [inline content-service/src/main/java/io/meeds/news/rest/NewsRest.java]: We should return a Response here IMO

File: content-service/src/main/java/io/meeds/news/service/impl/NewsServiceImpl.java [hunk 2]
```diff
@@ -838,6 +838,42 @@ public void shareNews(News news,
...
+        Calendar calendar = Calendar.getInstance();
+        metadataItem.setUpdatedDate(calendar.getTime().getTime());
+        metadataService.updateMetadataItem(metadataItem, Long.parseLong(identityManager.getOrCreateUserIdentity(versionCreator.getUserId()).getId()));
+      }
+      news.setIllustrationURL(NewsUtils.buildIllustrationUrl(existingPage.getProperties(), news.getLang()));
+      noteService.createVersionOfNote(existingPage, versionCreator.getUserId());
+      DraftPage draftPage = noteService.getLatestDraftPageByTargetPageAndLang(Long.parseLong(newsId), news.getLang());
+      while (draftPage != null) {
+        noteService.removeDraftById(draftPage.getId());
+        draftPage = noteService.getLatestDraftPageByTargetPageAndLang(Long.parseLong(newsId), news.getLang());
```
Comment (ahamdi) [inline content-service/src/main/java/io/meeds/news/service/impl/NewsServiceImpl.java]: Does this consider deleting all drafts created by all users for this page ?

File: content-service/src/main/java/io/meeds/news/rest/NewsRest.java (hunk 1 above)
Comment (sofyenne) [reply content-service/src/main/java/io/meeds/news/rest/NewsRest.java]: done!

File: content-service/src/main/java/io/meeds/news/service/impl/NewsServiceImpl.java (hunk 2 above)
Comment (sofyenne) [reply content-service/src/main/java/io/meeds/news/service/impl/NewsServiceImpl.java]: No need for the while loop with the concept of shared draft , the condition has been updated!

**Review by ahamdi — APPROVED** [review]

**Pull request merged**